src/emulation/mod.rs

```diff
@@ -137,18 +139,47 @@ impl SyscallMock {
     }
 }
 
+pub enum Executable {
+    Interpreter(PathBuf),
+    Program(PathBuf),
+}
+
+impl Deref for Executable {
+    type Target = Path;
+    fn deref(&self) -> &Path {
+        match self {
+            Executable::Interpreter(program) => &program,
+            Executable::Program(program) => &program,
+        }
+    }
+}
+
 pub fn run_against_protocol(
     context: Context,
-    executable: &Path,
+    interpreter: Option<Vec<u8>>,
+    program: &Path,
     expected: Protocol,
     unmocked_commands: &[Vec<u8>],
 ) -> R<TestResult> {
-    let syscall_mock = Tracer::run_against_mock(
-        executable,
-        expected.arguments.clone(),
-        expected.env.clone(),
-        |tracee_pid| SyscallMock::new(context, tracee_pid, expected, unmocked_commands),
-    )?;
+    let (executable, arguments): (Executable, Vec<String>) = match interpreter {
```

Comment:
initialize the `executable` and `arguments` values depending on whether we are running a script directly, or invoking it using an interpreter